SonarQube.TeamBuild.PreProcessor/SonarQubeServerFactory.cs

```diff
@@ -12,14 +12,18 @@ namespace SonarQube.TeamBuild.PreProcessor
 {
     internal class SonarQubeServerFactory : ISonarQubeServerFactory
     {
-        public ISonarQubeServer Create(ProcessedArgs args)
+        public ISonarQubeServer Create(ProcessedArgs args, ILogger logger)
         {
             if (args == null)
             {
                 throw new ArgumentNullException("args");
             }
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
 
-            return new SonarWebService(GetDownloader(args), args.GetSetting(SonarProperties.HostUrl));
+            return new SonarWebService(GetDownloader(args), args.GetSetting(SonarProperties.HostUrl), logger);
```

Comment:
The implementing class now expects a logger in the constructor, which means the factory method "Create" method needs to accept the logger parameter.
The SonarWebService class should probably do more diagnostic logging than it currently does. This change will make it easy to add logging without impacting the interfaces further.